pkg/segment/query/processor/searcher.go

```diff
@@ -40,6 +40,8 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+const segmentBatchOffset = 2 // TODO: Find an optimal value
```

Comment:
What does batch offset mean?

Reply:
addressed in the latest commit. Removed this offset to select cutOff timestamps. We are now getting cutOffs one by one from segments.